mhs-reference-implementation/mhs/transport/http_transport.py

```diff
@@ -0,0 +1,59 @@
+from pathlib import Path
+
+import requests
+
+
+class HttpTransport:
+    """A transport that supports the sending of messages via HTTP."""
+
+    def __init__(self, certs_dir):
+        """Create a new HttpTransport that loads certificates from the specified directory.
+        :param certs_dir: A string containing the path to the directory to load certificates from.
+        """
+        client_cert_path = str(Path(certs_dir) / "client.cert")
+        client_key_path = str(Path(certs_dir) / "client.key")
+        self._cert = (client_cert_path, client_key_path)
+        self._verify = str(Path(certs_dir) / "client.pem")
+
+    def make_request(self, interaction_details, message):
+        """Make a request for the specified interaction, containing the provided message. Raises an exception if a
+        non-success HTTP status code is returned by the server.
+
+        :param interaction_details: A dictionary containing details of the interaction this message is for.
+        :param message: The message body to send.
+        :return: The content of the response returned by the remote end.
+        """
+
+        url = interaction_details['url']
+
+        headers = HttpTransport._build_headers(interaction_details)
+
+        request_method = HttpTransport._get_request_method(interaction_details)
+
+        response = request_method(url,
+                                  data=message,
+                                  headers=headers,
+                                  cert=self._cert,
+                                  verify=self._verify)
+
+        response.raise_for_status()
```

Comment:
Just out of interest what does this do?

Reply:
It checks the status code of the response and raises an `HTTPError` if it's 4xx or 5xx. Worth a comment?

Reply:
I think that is pretty clear now. I don't think it needs a comment.